Bug 1881898: fix Quickstart subtitle block

abhinandan13jan opened the pull request:
# Addresses
https://issues.redhat.com/browse/ODC-4779

# Problem
The sub title of the main and step headers are misaligned when the header text is wrapping to another line

# Solution
- Put Step no inside h3 and within inline-block wrapper
- Put Duration inside Header and within inline-block wrapper

# Screenshots
![Screenshot from 2020-10-01 13-43-29](https://user-images.githubusercontent.com/24852534/94785632-8b0dc500-03ed-11eb-911a-53cd593b0564.png)
![Screenshot from 2020-10-01 13-42-57](https://user-images.githubusercontent.com/24852534/94785645-906b0f80-03ed-11eb-88cf-e59602c6aa2f.png)
![Screenshot from 2020-10-01 13-42-47](https://user-images.githubusercontent.com/24852534/94785663-9660f080-03ed-11eb-907c-f263907304cb.png)
![Screenshot from 2020-10-01 13-42-34](https://user-images.githubusercontent.com/24852534/94785687-9e209500-03ed-11eb-92c2-f01af2687c00.png)


# Tests
Altered tests QuickStartHeader.spec.scss

# Browser conformance

Chrome, Firefox

rohitkrai03 reviewed (changes requested):
The bug also mentions that the duration shown after the quick start title should not wrap around to next line. In your screenshots I only see the issue resolved for task titles.

abhinandan13jan commented:
> The bug also mentions that the duration shown after the quick start title should not wrap around to next line. In your screenshots I only see the issue resolved for task titles.

@rohitkrai03 I've added the Header part changes and also updated the screenshots. However, it looks a little odd in some cases.

abhinandan13jan commented:
/kind bug

rohitkrai03 commented:
> > The bug also mentions that the duration shown after the quick start title should not wrap around to next line. In your screenshots I only see the issue resolved for task titles.
> 
> @rohitkrai03 I've added the Header part changes and also updated the screenshots. However, it looks a little odd in some cases.

@abhinandan13jan Yeah, looking at the screenshots it seems the text is wrapping at wrong places. Not sure if using `<Text>` is a good idea here. @christianvogt WDYT?

christianvogt commented:
![image](https://user-images.githubusercontent.com/14068621/94709231-81b02a00-0313-11eb-9af6-d147ba673305.png)

Wrap the duration in an element with `display: inline-block`

christianvogt commented on frontend/packages/console-app/src/components/quick-starts/QuickStartDrawer.tsx:
```diff
@@ -78,10 +80,12 @@ const QuickStartDrawer: React.FC<QuickStartDrawerProps> = ({
-            <Title headingLevel="h6" size="md" className="text-secondary">
-              {`${quickStart?.spec.duration} minutes`}
+              {quickStart?.spec.displayName}{' '}
+              <span className="co-quick-start-drawer__duration">
+                <Text component={TextVariants.small} className="text-secondary">
+                  {`${quickStart?.spec.duration} minutes`}
+                </Text>
+              </span>
```
Why are you using `Text` ? Fairly certain `Text` is only useful inside of `TextContent`.

```suggestion
              <small className="co-quick-start-drawer__duration text-secondary">
                  {`${quickStart?.spec.duration} minutes`}
              </span>
```

abhinandan13jan replied:
updated

abhinandan13jan replied:
I was using Text as I simply moved a pre-existing block inside the Header without modifying it

christianvogt commented on frontend/packages/console-app/src/components/quick-starts/controller/QuickStartTaskHeader.tsx:
```diff
@@ -71,12 +71,13 @@ const QuickStartTaskHeader: React.FC<QuickStartTaskHeaderProps> = ({
         <TaskIcon taskIndex={taskIndex} taskStatus={taskStatus} isActiveTask={isActiveTask} />
         {title}
+        {isActiveTask && subtitle && (
+          <Text component={TextVariants.a} className="text-secondary">
+            {' '}
+            {subtitle}
+          </Text>
```
You created a class name never used it. Also you need to remove the space from the inline-block element.
```suggestion
          <>
            {' '}
            <small className="co-quick-start-task-header__subtitle text-secondary">
              {subtitle}
            </small>
          </>
```

abhinandan13jan replied:
updated

abhinandan13jan commented:
/bugzilla refresh

christianvogt commented:
/approve

rohitkrai03 reviewed (approved):
/lgtm

rohitkrai03 commented:
/retest
